io/net/virt-net/Network_virtualization_driver_bind.py

```diff
@@ -66,6 +66,9 @@ def test(self):
         """
         Performs driver unbind and bind for the Network virtualized device
         """
+        if not self.ping_check():
+            self.cancel("Please setup the IP configuration for your test interface and continue")
```

Comment:
@abdhaleegit Instead of canceling, can we set the IP for all interfaces and then start the test ?

Reply:
@narasimhan-v I had that thought in mind, but again what about peer ip setting ? should this be handled here ?

Reply:
only host IP.

Reply:
ok, in that case the the ip configuration done using configure_network utils vanishes after the device reset
so, incase of suse one has to set the IP configuration using yast2 tool or ifcg file, in case of redhat user has to create the ifcfg scripts for interface ip configuration.. this is must for my test to continue smoothly.. in case of configuration done using `ip commands` the ip setup is gone after ifup/down/bind/unbind/module load/unload/dlpar etc  and test will fail saying the bind/unbind resulted in network loss.. but actually it is config loss..

should script files has to be handled here for host interface ? @narasimhan-v

Reply:
@abdhaleegit The configuration should be done using ifcfg-files. Else, the IP configuration can be lost. Use avocado/utils/configure_network for the same.

Reply:
@narasimhan-v yes configure_network fails to create files for the vNIC interface

when ever a vnic/sriov interface is instantly added to linux,, there are no sysfs files gets created and configure_network fails with error saying 'there is no interface availabe'

here is a snippet from utils:
```
conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s
        if os.path.exists(conf_file):
            shutil.move(conf_file, conf_file+".backup")
        else:
            raise NWException("%s interface not available" % interface)  --> fails for sriov/vnic
```
do we need to modify the utils to create a network file if not exists ?

Reply:
planning to send this to utils 

```
diff --git a/avocado/utils/configure_network.py b/avocado/utils/configure_network.py
index 12b232c..2ca3da0 100644
--- a/avocado/utils/configure_network.py
+++ b/avocado/utils/configure_network.py
@@ -44,8 +44,6 @@ def set_ip(ipaddr, netmask, interface, interface_type=None):
         conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % interface
         if os.path.exists(conf_file):
             shutil.move(conf_file, conf_file+".backup")
-        else:
-            raise NWException("%s interface not available" % interface)
         with open(conf_file, "w") as network_conf:
             if interface_type is None:
                 interface_type = 'Ethernet'
@@ -71,8 +69,6 @@ def set_ip(ipaddr, netmask, interface, interface_type=None):
         conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
         if os.path.exists(conf_file):
             shutil.move(conf_file, conf_file+".backup")
-        else:
-            raise NWException("%s interface not available" % interface)
         with open(conf_file, "w") as network_conf:
             network_conf.write("IPADDR=%s \n" % ipaddr)
             network_conf.write("NETMASK='%s' \n" % netmask)
```

Reply:
@abdhaleegit so `vnic/sriov` case interface not available it self in sysfs path .it really strange, if that the case not sure we create the file as it will miss leading the usages.

Reply:
@smruti77 Is this case not handled yet in configure_network ?

Reply:
@PraveenPenguin @smruti77 I think checking sysfs is not the right way to confirm if interface is available or not

Reply:
I feel netifaces is the correct way

Reply:
@abdhaleegit ah ok, anyway I am refactoring this module Just wait for that  later you can send a fix for this, if you want to I can cover will cover this as well, let me know your thought

Reply:
> @smruti77 Is this case not handled yet in configure_network ?

@narasimhan-v its handle for both nic and infiniband

Reply:
@abdhaleegit netifaces is easier to check, is vNIC expected to create sysfs or not ?

Reply:
> @narasimhan-v yes configure_network fails to create files for the vNIC interface
> 
> when ever a vnic/sriov interface is instantly added to linux,, there are no sysfs files gets created and configure_network fails with error saying 'there is no interface availabe'
> 
> here is a snippet from utils:
> 
> ```
> conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s
>         if os.path.exists(conf_file):
>             shutil.move(conf_file, conf_file+".backup")
>         else:
>             raise NWException("%s interface not available" % interface)  --> fails for sriov/vnic
> ```
> 
> do we need to modify the utils to create a network file if not exists ?

@abdhaleegit this will fail for all interface if that file is not present in that directory,we need to create that file in that directory.

Reply:
@smruti77 I remember @vaishnavibhat telling me that we need to create an interface file if not existing previously, please check if that PR is sent already to avocado

Reply:
@smruti77 https://github.com/avocado-framework/avocado/pull/3367/files

Reply:
@narasimhan-v any interface, until not configured(yast or file) is not expected to show up in sysfs

Reply:
@PraveenPenguin include this one too

Reply:
@narasimhan-v ok, need to add "shutil.move(conf_file, conf_file+".backup")" this also otherwise its fail in unset_ip.

Reply:
Sure